### ISSUE TITLE
Add a secondary background color to code blocks

### ISSUE THREAD

**Issue by JarWarren** [twentyhq/twenty#6029]
## Add a secondary background color to inline code blocks

Can we change the background color of `code blocks` (backticked text) to make it more visibly distinct? The website has this, but the CRM doesn't.

## Current behavior

Currently the font and kerning change, but it's easy to miss. The following screenshot contains a code block. This affects both light and dark mode.

<img width="435" alt="Screenshot 2024-06-25 at 7 42 21 PM" src="https://github.com/twentyhq/twenty/assets/43893126/27ce0716-06ce-4057-a0cc-e8f3df26d473">

## Expected behavior

Secondary background color, consistent with the website.

<img width="550" alt="Screenshot 2024-06-25 at 7 45 53 PM" src="https://github.com/twentyhq/twenty/assets/43893126/27dff53f-00ec-41c3-8d5a-e0ca857b50c9">

```[tasklist]
### Tasks
- [ ] https://github.com/twentyhq/twenty/issues/6929
- [ ] https://github.com/twentyhq/twenty/issues/6930
```

**Comment by Bonapara** [comment]
> The following screenshot contains a code block. This affects both light and dark mode.

Hi @JarWarren, thanks for the feedback! Where is the code block in your screenshot? In the title?

**Comment by JarWarren** [comment]
Package.json

**Comment by Bonapara** [comment]
I don't understand what you mean, @JarWarren. When you say "the following screenshot contains a code block," where should it be located visually within the screenshot?

**Comment by JarWarren** [comment]
Sorry, I should have been more specific in the original description. The Sample Task screenshot has a code block around `package.json` (if you look closely), but code blocks are so visually similar to other text that it's easy to miss them.

Using backticks in Twenty renders the backticked text within an inlined `<code>` html tag, which is perfect, but I'm basically suggesting we change the CSS so they pop out more.

This screenshot and the original one both have a `<code>` tag around `package.json`.

<img width="1020" alt="css" src="https://github.com/twentyhq/twenty/assets/43893126/75244fd2-dec1-420e-998e-2b6ad84fe28a">

**Comment by Bonapara** [comment]
Oh got you, I was looking at inline fields. Thanks!

**Comment by Bonapara** [comment]
The border should be: `medium`
The background should be: `transparent light`
Color should be: `Tag/red`

(Thanks for the design @JarWarren ;) )

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara To write code shouldn't there be an option in this menu of ```code```.
<img width="229" alt="Screenshot 2024-07-30 at 11 04 30 PM" src="https://github.com/user-attachments/assets/34bdbbdf-ca48-4feb-a1a4-5c59def986cb">
So that we can then add styles to it .

**Comment by Bonapara** [comment]
Yes it could @Faisal-imtiyaz123, great idea! 
The icon should be: https://tabler.io/icons/icon/code

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara Could you please assign me.

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara There is a problem though, how to get out of `code mode`.Maybe we could add ```code``` option in the formatting toolbar which appears when some text is selected.

**Comment by BalanaguYashwanth** [comment]
Hello @Faisal-imtiyaz123 @Bonapara,

Are you working on this issue ?

**Comment by Faisal-imtiyaz123** [comment]
@BalanaguYashwanth Yes. I am awaiting @Bonapara's reply on this one.

**Comment by Bonapara** [comment]
Hi @Faisal-imtiyaz123, well noticed. To quit the code mode the user should press "Arrow right"  or "Enter" (that would create a new block)

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara But the `code` shouldn't be a block right? Should it  rather  not be an  an `inline` element ?

**Comment by Bonapara** [comment]
Yes talking about inline code when I say "To quit the code mode the user should press "Arrow right" or "Enter" (that would create a new block)"

Are the `code snippets` you're mentioning multi-line code blocks? If yes they are complementary.

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara Sorry for asking too much but I would like to clear everything first. The current behaviour with pressing `Enter` or `Arrow right` is that the cursor moves to a `new line` . But for `code` pressing `Enter` or `Arrow right` should create a `code snippet` while remaining in the same line. Did i get it correct?

**Comment by Bonapara** [comment]
To clarify, I think we should do two separate things: 1) Adapt the inline code style. 2) Optionally add a code block type.

### 1) Inline Code Style

When you press "cmd+E" today, the selected text switches to inline code with a monospace font but no background. We want to correct this style as follows:

-  Border: medium
-  Background: transparent light
-  Color: Tag/red

The behavior of → and ⏎ described above applies to this feature.

https://github.com/user-attachments/assets/f2e4f83b-9dd5-4d9d-b8bc-7cfb4ae4870b

### 2) Multiline Code Block

We can add a code block option for multi-line code. The behavior should be the same as in Notion.

![CleanShot 2024-09-05 at 14 11 06](https://github.com/user-attachments/assets/5fdea709-6b54-478b-8e46-52f6ec3f20b8)

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara Got it perfectly! Thank you.

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara There is a package for block note which let's you to create code blocks with proper `syntax highlighting`. Please see if we should use it. https://github.com/defensestation/blocknote-code

**Comment by Bonapara** [comment]
@Faisal-imtiyaz123, thanks! The issue with this library is that it doesn't rely on `Monaco`, which we chose as our code editor. We don't want to introduce another library to the repository. So, we either find a library that relies on Monaco or stick to the inline code block!

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara I figured out the `inline code snippets`. I was asking the above for the `multi line code blocks`. This library is just a `plugin` for the currently used `block note editor` to create multi line code blocks. I think what you meant to say was that you don't like the way it does `syntax highlighting` and want to use the style of `monaco-react`. If so, in that case I would try making a `multi line code block` with the `syntax highlighting` provided by `monaco-react`.

**Comment by Bonapara** [comment]
Hi @Faisal-imtiyaz123, it's not a question of style but of dependency. Blocknote-code relies on `codemirror`, whereas we are using Monaco for our `Functions` functionnality. This is to avoid having two different code editor dependencies in the code!

![CleanShot 2024-09-06 at 17 02 14](https://github.com/user-attachments/assets/b4ec26ae-6d8c-4abc-9d4a-e7d1e928c041)

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara  I am sorry for troubling you a lot here!. I deleted that comment though after realising what you meant. Github bug i guess lol..

**Comment by Bonapara** [comment]
No worries, it's better to be on the same page! Thanks for contributing @Faisal-imtiyaz123

**Comment by Faisal-imtiyaz123** [comment]
@Bonapara I have made a PR for the inline code part. As for the `code block` I tried integrating `monaco editor` as a block but encountered a weird bug there. It would sometimes write into the `editor` and sometimes whatever we try to write int the `editor`, that is written after or before the `editor` and not `inside it`. I even encountered it in the `code sandbox` example on `monaco react npm page`. Still I will try to see if i can figure something out.
